R/amend_large_polygons.R

```diff
@@ -0,0 +1,173 @@
+#' Extract large polygons and assign additional layers
+#'
+#'Extract large polygons and adjacent polygons,
+#'Cut in CEF glaciers and FWA wetlands and lakes
+#'Merge BEM and correct assigned BEU/SDEC
+#'
+#'@param vri_bem VRI-BEM feature class
+#'@param glaciers sf
+#'@param lakes sf
+#'@param wetlands sf
+#'@param bem sf
+#'
+#'@import sf
+#'@export
+
+#run the xl polygons functions and merge with the regular vri-bem
+
+amend_large_polygons <- function (vri_bem, glaciers, lakes, wetlands, bem){
+
+vri_bem <- vri_bem %>% dplyr::mutate(area = st_area(.))
+
+if(max(vri_bem$area) > units::set_units(35000000,m^2)){
+
+#Extract large polygons
+vrixl <- extract_large_polygons(vri_bem)
+
+#Merge additional layers
+vrixl_merge <- merge_extra_layers(vrixl, glaciers, lakes, wetlands)
+
+#Merge BEM on XL polygons
+vri_bem_xl <- merge_bem_on_vri_xl(vrixl_merge, bem)
+
+#Remove XL polygons from initial VRI-BEM and replace with updated polygons
+vri_bem_xl_diff <- st_difference(vri_bem, st_union(st_geometry(vri_bem_xl))) |> st_make_valid() |>
+  st_collection_extract() |> st_cast("POLYGON", warn = FALSE) |> st_make_valid()
+
+#Make sure TEIS_ID is character
+vri_bem_xl_diff <- mutate(vri_bem_xl_diff, TEIS_ID = as.character(TEIS_ID))
+vri_bem_xl <- mutate(vri_bem_xl,TEIS_ID = as.character(TEIS_ID))
+
+#Bind vri-bem and xl polygons
+vri_bem <- dplyr::bind_rows(vri_bem_xl_diff, vri_bem_xl)
+
+return(vri_bem)
+}
+else{
+  return(vri_bem)
+}
+}
+
+
+
+#extract large polygons
+extract_large_polygons <- function (vri_bem) {
+
+  #find polygons ovre 3500 ha
+  large_polygons <- vri_bem %>%
```

Comment:
Just a suggestion but R now has a pipe operator included in base R  |> instead of %>%